src/main/java/net/smartcosmos/extension/tenant/config/AnonymousAccessSecurityConfiguration.java

```diff
@@ -1,36 +1,39 @@
 package net.smartcosmos.extension.tenant.config;
 
+import lombok.extern.slf4j.Slf4j;
+
 import org.springframework.boot.autoconfigure.security.SecurityProperties;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
 import org.springframework.core.annotation.Order;
 import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
 
-import net.smartcosmos.extension.tenant.auth.SmartCosmosAnonymousUser;
+import net.smartcosmos.security.user.SmartCosmosUserConfiguration;
 
 @Configuration
-@Order(SecurityProperties.DEFAULT_FILTER_ORDER)
-public class AnonymousAccessSecurityConfiguration extends WebSecurityConfigurerAdapter {
+// Put this in order before the typical SMART COSMOS Security
+@Order(SecurityProperties.ACCESS_OVERRIDE_ORDER - 2)
+@Slf4j
+@EnableResourceServer
+@Import(SmartCosmosUserConfiguration.class)
+public class AnonymousAccessSecurityConfiguration extends ResourceServerConfigurerAdapter {
 
     @Override
-    protected void configure(HttpSecurity http) throws Exception {
+    public void configure(HttpSecurity http) throws Exception {
 
-        http.requestMatchers()
-            .antMatchers("/tenants/**")
-            .and()
-            .authorizeRequests()
+        log.debug("Overriding default security behavior.");
+        http.authorizeRequests()
             .antMatchers(HttpMethod.POST, "/tenants")
             .permitAll()
             .and()
+            .antMatcher("/**")
             .authorizeRequests()
             .anyRequest()
             .authenticated()
             .and()
-            .anonymous()
```

Comment:
Removing this will let us fall back to the default Spring anonymous user (if any), which actually adds complexity, or at least inconsistency, because otherwise each and every principal getting to the endpoints will be an instance of `SmartCosmosUser`.